ansible_base/lib/abstract_models/common.py

```diff
@@ -59,14 +60,36 @@ def save(self, *args, **kwargs):
         '''
         This save function will provide the following features automatically.
           * It will automatically add the modified fields for changing items
+
+        There are several edge cases to be aware of:
+          * If the user is logging in and their "last_login" field is getting
+            changed (in their user model instance), then we do not update
+            "modified_by". (This is so they can see if someone changed their
+            user object while they were away).
+          * If no fields in update_fields are "editable", then we don't update
+            "modified_by".
+          * "modified" is updated in every case, since it's an auto_now
+            timestamp.
         '''
+        has_update_fields = kwargs.get('update_fields') is not None
         update_fields = list(kwargs.get('update_fields', []))
-
-        if 'modified_by' not in update_fields:
+        is_user_logging_in = isinstance(self, AbstractUser) and update_fields == ['last_login']
+        has_editable_field = any(self._meta.get_field(field).editable for field in update_fields)
+
+        if 'modified_by' in update_fields:
+            # Explicit update of modified_by, don't mess with it
+            pass
+        elif is_user_logging_in:
+            # User is logging in, only last_login is changing, don't update modified_by
+            pass
+        elif has_update_fields and not has_editable_field:
+            # No editable fields are changing, don't update modified_by
+            pass
+        else:
             self.modified_by = current_user_or_system_user()
             update_fields.append('modified_by')
 
-        if kwargs.get('update_fields') is not None:
+        if has_update_fields:
```

Comment:
Probably non-functional suggestion:

If `modified_by` _was not set_ then I would prefer not to make edits to `kwargs` because that would get us involved in other error paths we don't want to touch.

This may be accomplished by simply increasing the indent of this `if` by one tab.